diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22985383..15732b53 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,7 +143,7 @@ target_include_directories(franka PUBLIC
 target_link_libraries(franka PRIVATE
   Poco::Foundation
   Poco::Net
-  Eigen3::Eigen3
+  Eigen3::Eigen
   Threads::Threads
   pinocchio::pinocchio
 )
diff --git a/cmake/FindEigen3.cmake b/cmake/FindEigen3.cmake
index dc2fb7c9..b4c230d6 100644
--- a/cmake/FindEigen3.cmake
+++ b/cmake/FindEigen3.cmake
@@ -1,16 +1,7 @@
-find_package(Eigen3 CONFIG)
-mark_as_advanced(FORCE Eigen3_DIR)
+find_package(Eigen3 REQUIRED CONFIG)

-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(Eigen3
-  FOUND_VAR Eigen3_FOUND
-  REQUIRED_VARS EIGEN3_INCLUDE_DIRS
-)
+mark_as_advanced(FORCE Eigen3_DIR)

-if(NOT TARGET Eigen3::Eigen3)
-  add_library(Eigen3::Eigen3 INTERFACE IMPORTED)
-  set_target_properties(Eigen3::Eigen3 PROPERTIES
-    INTERFACE_INCLUDE_DIRECTORIES ${EIGEN3_INCLUDE_DIRS}
-    INTERFACE_COMPILE_DEFINITIONS "${EIGEN3_DEFINITIONS}"
-  )
+if(NOT TARGET Eigen3::Eigen)
+  message(FATAL_ERROR "Eigen3 was found, but target Eigen3::Eigen is not available")
 endif()
